drivers/hmis/app/models/hmis/workflow_execution/engine.rb

```diff
@@ -72,24 +72,26 @@ def next_task_steps(step)
       nodes.map { |node| steps_by_node_id[node.id] }.compact
     end
 
-    def send_message(...)
+    def send_message(user, ...)
       message = Hmis::WorkflowExecution::Message.new(...)
-      message_handler.call(message)
+      message_handler.call(message, user)
     end
 
-    def process_triggers(node, event_type)
+    def process_triggers(node, event_type, user, step = nil)
```

Comment:
This is becoming a lot of positional arguments to keep track of, I'd suggesting switching to keyword args. 

I am a little confused about needing to pass both `node` and `step` since they have a relationship to each other. Can the step be accessed from the node instead?
```
step = node.step if node.task? # if this is a task node, it should(?) have an associated step
```

Reply:
Ah of course, I see. So sometimes we want to pass the `step` but sometimes we only want to pass the `node` (which may not even _have_ a step). I think your adjustment to the engine makes sense to me.